doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1593,6 +1594,31 @@ Number notations
          function application, constructors, inductive type families,
          sorts, and primitive integers) will be considered for printing.
 
+      .. _number-string-via:
+
+      :n:`via @qualid__ind using ( {+, @qualid__constant @qualid__constructor } )`
+         When using this option, :n:`@qualid__type` no
+         longer needs to be an inductive and is mapped to the
+         inductive :n:`@qualid__ind` according to the provided
+         list of pairs, whose first component :n:`@qualid__constant` is a
+         constant of type :n:`@qualid__type`
+         (or a function of type :n:`{* _ -> } @qualid__type`) and the second a
+         constructor of type :n:`@qualid__ind`. The type
+         :n:`@qualid__type` is then replaced by :n:`@qualid__ind` in the
+         above parser and printer types.
+
+         When :n:`@qualid__constant` is surrounded by square brackets,
+         the implicit arguments of :n:`@qualid__constant` are ignored
```

Comment:
Please specify whether this computation is performed at the numeral notation declaration time or whether it is performed when the notation is used.  (I think it should be the first; it seems bad for `Arguments` to be able to break a numeral notation.)  Please also clarify if there's any difference between maximally inserted implicits and non-maximally-inserted implicits, how constants with multiple options for implicit arguments are treated (such as `eq_refl` in `Arguments eq_refl {A x} , [A] x`), and whether only the leading implicits are considered, or if all implicits are considered.